packages/next/next-server/server/utils.ts

```diff
@@ -14,3 +15,28 @@ export function cleanAmpPath(pathname: string): string {
   pathname = pathname.replace(/\?$/, '')
   return pathname
 }
+
+export function collectEnv(page: string, pageEnv?: string[]): Env {
+  const missingEnvKeys = new Set()
+  const collected = pageEnv
+    ? pageEnv.reduce((prev: Env, key): Env => {
+        if (process.env[key]) {
```

Comment:
We should probably differentiate between missing and empty, i.e. `undefined` vs `''`. Users may explicitly be setting an empty value.

Reply:
Updated